### PR TITLE
feat: Set internal access on GCP to prevent hacking attempts

### LINKED ISSUE #192: Set internal access on GCP to prevent hacking attempts
**Problem**

There are many (unsuccessful) hacking attempts to access our PostGres database that are indirectly impacting performance. 

**Possible Solution**

Set internal access only to our PostgreSQL database on GCP to prevent so many hacking attempts.

### PR DESCRIPTION
**Pull request by davidgamez**
## Summary:

This PR updates the database deployed to GCP, removing the public IP address in order to protect it from external login attempts. To achieve connection with internal GCP resources, all functions needed to connect to DB are attached to the default VPC. To connect from GH actions, a zerotier _bastion_  VM is deployed to each environment. 
List of major changes:
- Remove public IP from PROD and PRE-PROD databases
- Attach VPC to API functions
- Attach VPC to batch functions

List of minor changes:
- Updated checkout and google auth actions to the newer version on _touched_ files
- Using local installed liquibase instead of action due to issues connecting to GH actions localhost
- Ensure zerotier machine is running before connecting to DB
- Fixed PROD DB deploy workflow
- Removed deployment of DEV database workflow

Closes #192 

**Testing tips:**

- Run smoke test in DEV and QA API and batch functions

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: .github/workflows/db-update-qa.yml [hunk 1]
```diff
@@ -18,31 +18,69 @@ jobs:
...
+        uses: google-github-actions/setup-gcloud@v2
 
       - name: Get Database Instance IP
         id: get_ip
         run: |
           gcloud config set project ${{ vars.QA_MOBILITY_FEEDS_PROJECT_ID }}
-          DB_IP=$(gcloud sql instances describe ${{ secrets.DB_INSTANCE_NAME }} --format=json | jq -r '.ipAddresses[] | select(.type=="PRIMARY") | .ipAddress')
+          DB_IP=$(gcloud sql instances describe ${{ secrets.DB_INSTANCE_NAME }} --format=json | jq -r '.ipAddresses[] | select(.type=="PRIVATE") | .ipAddress')
           echo "DB_IP=$DB_IP" >> "$GITHUB_OUTPUT"
+  gw_status:
```
Comment (cka-y) [inline .github/workflows/db-update-qa.yml:37]: suggestion: extract the `gw_status` as a separate callable workflow

File: infra/postgresql/main.tf [hunk 2]
```diff
@@ -32,14 +57,11 @@ resource "google_sql_database_instance" "db" {
...
     }
     ip_configuration {
-      ipv4_enabled = true
-
-      authorized_networks {
-        name  = "all-ips"
-        value = "0.0.0.0/0"
-      }
+      ipv4_enabled = false
+      private_network = "projects/${var.project_id}/global/networks/default"
```
Comment (cka-y) [inline infra/postgresql/main.tf:61]: 🎉

**Review by cka-y — APPROVED** [review]
🤩

File: .github/workflows/db-update-qa.yml (hunk 1 above)
Comment (davidgamez) [reply .github/workflows/db-update-qa.yml:37]: Let's do this in the future to unblock all other work waiting for this change

**Pull request merged**